arangod/Graph/PathEnumerator.cpp

```diff
@@ -51,6 +51,24 @@ void PathEnumerator::setStartVertex(arangodb::velocypack::StringRef startVertex)
   TRI_ASSERT(_enumeratedPath.vertices.size() == 1);
 }
 
+bool PathEnumerator::keepEdge(arangodb::graph::EdgeDocumentToken& eid,
+                              arangodb::velocypack::Slice edge,
+                              velocypack::StringRef sourceVertex, size_t depth,
+                              size_t cursorId) {
+  if (_opts->hasEdgeFilter(depth, cursorId)) {
+    VPackSlice e = edge;
+    if (edge.isString()) {
+      e = _opts->cache()->lookupToken(eid);
+    }
+    if (!_traverser->edgeMatchesConditions(e, sourceVertex, depth, cursorId)) {
+      // This edge does not pass the filtering
+      return false;
+    }
+  }
+
+  return _opts->destinationCollectionAllowed(edge, sourceVertex);
```

Comment:
Is that correct now in case we have a collection restriction but no edge filter?
I think the `destinationCollectionAllowed()` method requires `edge` to be an object, but it will be turned into an object only in lines 60-62 in case there is an edge filter available.